File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/bot/BumblebeeConfig.java
package com.github.bumblebee.bot;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

import java.util.HashMap;
import java.util.Map;

@Configuration
@ConfigurationProperties(prefix = "bumblebee")
public class BumblebeeConfig {

    private String token;

    private String url;

    private String certificatePath;

    private Map<String, String> commands = new HashMap<>();

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getCertificatePath() {
        return certificatePath;
    }

    public void setCertificatePath(String certificatePath) {
        this.certificatePath = certificatePath;
    }

    public Map<String, String> getCommands() {
        return commands;
    }
}

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/bot/BumblebeeConfig.kt
package com.github.bumblebee.bot

import com.github.telegram.api.BotApi
import com.github.telegram.api.TelegramBot
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

data class WebhookConfig(var enabled: Boolean = false)

@Configuration
@ConfigurationProperties(prefix = "bumblebee")
class BumblebeeConfig {
    lateinit var token: String
    var url: String = ""
    var certificatePath: String? = null
    var commands: Map<String, String> = HashMap()
    var webhook: WebhookConfig = WebhookConfig()

    @Bean
    fun telegramApi(): BotApi = TelegramBot.create(token)
}
